lib/engine/step/g_1848/dutch_auction.rb

```diff
@@ -0,0 +1,142 @@
+# frozen_string_literal: true
+
+require_relative '../base'
+require_relative '../auctioner'
+
+module Engine
+  module Step
+    module G1848
+      class DutchAuction < Base
+        include Auctioner
+
+        attr_reader :companies
+
+        # On your turn, must buy any private or reduce price of any private
+        # Once you own a private, you may pass
+        # if everything is reduced to minimum price, and nobody has bought a private, you must buy one
+        # if any privates are left and everyone passes in a row, owned privates pay and auction continues
+        # if someone has bought a private and everything is minimum, you can pass
+
+        ACTIONS = %w[bid reduce].freeze
+        ACTIONS_WITH_PASS = %w[bid reduce pass].freeze
+
+        MIN_PRICES = {
+          'P1' => 0,
+          'P2' => 40,
+          'P3' => 80,
+          'P4' => 140,
+          'P5' => 140,
+          'P6' => 200,
+        }.freeze
+
+        def description
+          "Buy a Company or Reduce its Price by #{@game.format_currency(5)}"
+        end
+
+        def available
+          @companies
+        end
+
+        def auctioneer?
+          false
+        end
+
+        def committed_cash(_player, _show_hidden = false)
+          0
+        end
+
+        def pass_description
+          'Pass'
+        end
+
+        def process_bid(action)
+          company = action.company
+          price = company.min_bid
+          buy_company(current_entity, company, price)
+          @round.next_entity_index!
+        end
+
+        def process_pass(action)
+          entity = action.entity
+          @log << "#{entity.name} passes"
+          entity.pass!
+          all_passed! if entities.all?(&:passed?)
+          @round.next_entity_index!
+        end
+
+        def process_reduce(action)
```

Comment:
you'll need to change this to process_assign

Reply:
bca96f79017673bb5f853eb2899a76f486a7d7b8